nbconvert/exporters/tests/test_templateexporter.py

```diff
@@ -114,6 +117,46 @@ def test_relative_template_file(self):
             exporter = self._make_exporter(config=config)
             assert os.path.abspath(exporter.template.filename) == template
             assert os.path.dirname(template) in [ os.path.abspath(d) for d in exporter.template_path ]
+    
+    def test_in_memory_template(self):
+        # Loads in an in memory template using jinja2.DictLoader
+        # creates a class that uses this template with the template_file argument
+        # converts an empty notebook using this mechanism
+        my_loader = DictLoader({'my_template': "{%- extends 'rst.tpl' -%}"})
+        
+        class MyExporter(TemplateExporter):
+            template_file = 'my_template'
+        
+        exporter = MyExporter(extra_loaders=[my_loader])
+        nb = v4.new_notebook()
+        out, resources = exporter.from_notebook_node(nb)
+
+
+    @pytest.mark.xfail(strict=True, raises=TemplateNotFound)
+    def test_in_memory_template_failure_to_find(self):
+        
+        # Create a class with a custom template. Opens a temporary diectory, in
+        # which it writes a file at the appropriate location to be found in
+        # that template. Creates a class instance and then then it manually
+        # adds the template extension (erroneously) to the template file name,
+        # and tries to convert an empty notebook using this mechanism (at which
+        # point it should fail).
+        
+        class MyExporter(TemplateExporter):
+            self.template_file = 'my_template.tpl'
+        
+        with tempdir.TemporaryDirectory() as td:
+            template = os.path.join(td, 'my_template.tpl')
+            test_output = 'absolute!'
+            with open(template, 'w') as f:
+                f.write(test_output)
+            exporter = MyExporter()
+            exporter.template_file = template + exporter.template_extension
```

Comment:
I think we can simplify this a bit. All we need to test it, I think, is a template name that we can be reasonably sure won't exist, like `does_not_exist.tpl`. Then we don't need to subclass TemplateExporter, or create a temporary directory, or create a template that does exist. We can just instantiate `TemplateExporter(template_file='does_not_exist.tpl')` and check that trying to use it raises TemplateNotFound.